src/ferny/ssh_askpass.py

```diff
@@ -0,0 +1,200 @@
+import logging
+import re
+from typing import ClassVar, Match, Optional, Sequence
+
+from .interaction_agent import AskpassHandler
+
+logger = logging.getLogger(__name__)
+
+
+class AskpassPrompt:
+    """An askpass prompt resulting from a call to ferny-askpass.
+
+      stderr: the contents of stderr from before ferny-askpass was called.
+              Likely related to previous failed operations.
+      messages: all but the last line of the prompt as handed to ferny-askpass.
+                Usually contains context about the question.
+      prompt: the last line handed to ferny-askpass.  The prompt itself.
+    """
+    stderr: str
+    messages: str
+    prompt: str
+
+    def __init__(self, prompt: str, messages: str, stderr: str) -> None:
+        self.stderr = stderr
+        self.messages = messages
+        self.prompt = prompt
+
+    def reply(self, response: str) -> None:
+        pass
+
+    def close(self) -> None:
+        pass
+
+    async def handle_via(self, responder: 'SshAskpassResponder') -> None:
```

Comment:
Why the stringified type? Is that because that class only gets declared later in this file? Or doesn't Py 3.6 understand proper classes yet?

Reply:
It's exactly this.  There are three ways that type annotations are interpreted:

 - with `from __future__ import annotations` type annotations are all essentially treated as strings.  The idea is that you can parse the strings later, given the right information about the namespace they were defined in.  This was expected to one day become the default, but it didn't happen.  Still, it would be a reasonable approach, but it's unavailable on Python 3.6.  It's equivalent to quoting all of your type annotations (as we do here).
 - without `from __future__ import annotations` type annotations are parsed as they are encountered.  This means that references to types that haven't been defined yet don't work.  Unfortunately that's where we're stuck, but we can use quotes to escape issues as they arise.
 - from Python 3.13 ([PEP 649](https://peps.python.org/pep-0649/) annotations are essentially created as functions which can be called later, which means the get the normal scoping rules from the rest of the language and can refer to things defined later in the file.  This is really nice, but nothing that we can use any time soon.